### PR TITLE
Fix collection bugs from #372

### LINKED ISSUE #373: Issue with contrast adjustment update when switching between datasets
When I switch between datasets, it seems that the contrast settings are kept for each dataset, but the sliders and the histogram do not update when I switch between datasets. It's doing the right thing, but not reporting the right thing in the histogram.

image

### LINKED ISSUE #374: Issue with missing channel after clicking Configuration
I clicked configuration here:
image

It brought up the datasets. When I clicked on either dataset, though, it doesn't show the brightfield channel anymore:
image

I also noticed that in the Configuration page, it doesn't show anything in the "Brightfield" "Channel" (just an empty field). Not sure if that is related:
image

### LINKED ISSUE #375: "Unknown dataset" flashes when changing Z
When I click a new position in Z (before caching), it takes a LONG time to load, and also shows the following:

image

After a while, this comes back:

image

### PR DESCRIPTION
**Pull request by bruyeret**
Fix multiple bugs introduced by PR #372:

Closes #373 
Closes #374 
Closes #375

### CODE REVIEW

**Comment by arjunrajlab** [comment]
Tested it out, looks good to me!

**Comment by arjunrajlab** [comment]
Z scrolling is less smooth than it used to be. Is it running some sort of check now? Unclear why that would be slower.

File: src/store/annotation.ts [hunk 1]
```diff
@@ -212,6 +212,20 @@ export class Annotations extends VuexModule {
 
   @Mutation
   public setAnnotations(values: IAnnotation[]) {
+    const nAnnotations = values.length;
+    // Check if annotations are the same
+    if (nAnnotations === this.annotations.length) {
+      let equals = true;
+      for (let i = 0; i < nAnnotations; ++i) {
+        if (values[i].id !== this.annotations[i].id) {
```
[the remark this answers is not in the document]
Comment (bruyeret) [reply src/store/annotation.ts:220]: I thought about it, but using `.every` is slower
Performance is important for this function as there can be a lot of annotations

image

**Comment by luciemac** [comment]
LGTM

**Pull request closed without merging**